indra/assemblers/pybel/assembler.py

```diff
@@ -589,7 +589,9 @@ def _get_evidence(evidence):
                     pc.CITATION_IDENTIFIER: cit_ref_str}
     pybel_ev[pc.CITATION] = citation
 
-    annotations = {}
+    annotations = {
+        'source_hash': evidence.get_source_hash(),
```

Comment:
All evidences have source hashes, regardless of if there's a source API or source identifier